src/tables/vhea.js

```diff
@@ -2,7 +2,8 @@ import * as r from 'restructure';
 
 // Vertical Header Table
 export default new r.Struct({
-  version:                r.uint16,  // Version number of the Vertical Header Table
+  majorVersion:           r.uint16,  // Major version number of the Vertical Header Table
+  minorVersion:           r.uint16,  // Minor version number of the Vertical Header Table
```

Comment:
Note that the version is a 16dot16 number, which is not the same as two uint16. The first number is a uint16, but the second number uses a _very_ different format, only values `0000`, `1000`, ..., `9000` are valid, and represent minor versions 0 through 9.

Reply:
Good day! 👋 Thank you for the clarification. As I see it, the `maxp` table uses the same approach (16dot16) to versioning. In this library, `maxp` describes the version as a simple 32-bit number with no additional major/minor divisions, because version, as I understand it, is not used anywhere else. https://github.com/foliojs/fontkit/blob/a5fe0a1834241dbc6eb02beea3b7414c118c5ac9/src/tables/maxp.js#L5 Should I apply the same approach here?

Reply:
Correct, there are three tables (`maxp`, `post`, `vhea`) that use 16dot16 for historical reasons, and thankfully no future table will ever be allowed to use it again. If the `maxp` table uses a 32 int, that may need fixing too, as there are technically two versions for the `maxp` table (0.5 and 1.0).

Reply:
Having said that, note that there may be some magic going on in `r.VersionedStruct` because the `post` table also uses 16dot16, which gets read as an int32 but gets turned into a "real" version anyway (looking at https://github.com/foliojs/fontkit/tree/master/src/tables/post.js)

Reply:
oh that's fun. You might have to implement a custom type for that. [Here](https://github.com/foliojs/restructure/blob/master/src/Number.js#L57-L70) is how `Fixed` is implemented. You'd need to do something similar to handle this type.

I would also not switch this to two separate fields as that might be considered a breaking change (if someone was using the `version` field before).

Reply:
I made a small encoder/decoder that should handle the given type and applied it to all 3 tables. A small note regarding the `Base` class of the restructure library: it is not exported, so we had to abandon inheritance and add the `fromBuffer` and `toBuffer` methods directly to `Version16Dot16`. The test results did not change in any way (at least no degradation was noticed).